var/spack/repos/builtin/packages/py-pycairo/package.py

```diff
@@ -13,18 +13,22 @@ class PyPycairo(PythonPackage):
 
     homepage = "https://www.cairographics.org/pycairo/"
     pypi = "pycairo/pycairo-1.17.1.tar.gz"
+    git = "https://github.com/pygobject/pycairo.git"
 
+    version("1.24.0", sha256="1444d52f1bb4cc79a4a0c0fe2ccec4bd78ff885ab01ebe1c0f637d8392bcafb6")
     version("1.20.0", sha256="5695a10cb7f9ae0d01f665b56602a845b0a8cb17e2123bfece10c2e58552468c")
     version("1.18.1", sha256="70172e58b6bad7572a3518c26729b074acdde15e6fee6cbab6d3528ad552b786")
     version("1.17.1", sha256="0f0a35ec923d87bc495f6753b1e540fd046d95db56a35250c44089fbce03b698")
 
-    depends_on("cairo@1.15.10: +pdf", when="@1.20.0:")
-    depends_on("cairo@1.13.1: +pdf", when="@:1.18.1")
-    depends_on("pkgconfig", type="build")
-    depends_on("py-setuptools", type="build")
-    depends_on("python@2.7:2.8,3.3:", when="@:1.17.1", type=("build", "run"))
+    depends_on("python@3.8:", when="@1.23:", type=("build", "run"))
+    depends_on("python@3.6:3", when="@1.20:", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("python@3.6:3", when="@1.20:1.22", type=("build", "run"))
```
Otherwise constraint will be applied to newer versions too.